targets/XPlatCppSdk/source/code/include/playfab/PlayFabIOSHttpPlugin.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <playfab/PlayFabCallRequestContainer.h>
+#include <playfab/PlayFabPluginManager.h>
+#include <thread>
+#include <mutex>
+#include <atomic>
+
+namespace PlayFab
+{
+    /// <summary>
+    /// PlayFabIOSHttpPlugin is an https implementation to interact with PlayFab services using IOSHTTP.
+    /// </summary>
+    class PlayFabIOSHttpPlugin : public IPlayFabHttpPlugin
+    {
+    public:
+        PlayFabIOSHttpPlugin();
+        PlayFabIOSHttpPlugin(const PlayFabIOSHttpPlugin& other) = delete;
+        PlayFabIOSHttpPlugin(PlayFabIOSHttpPlugin&& other) = delete;
+        PlayFabIOSHttpPlugin& operator=(PlayFabIOSHttpPlugin&& other) = delete;
+        virtual ~PlayFabIOSHttpPlugin();
+
+        virtual void MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer) override;
+        virtual size_t Update() override;
+
+    protected:
+        struct RequestTask;
+
+        void WorkerThread();
+        void ExecuteRequest(RequestTask& requestTask);
+
+        virtual std::string GetUrl(RequestTask& requestTask) const;
+        virtual void SetPredefinedHeaders(RequestTask& requestTask, void* urlRequest);
+        virtual bool GetBinaryPayload(RequestTask& requestTask, void*& payload, size_t& payloadSize) const;
+        virtual void ProcessResponse(RequestTask& requestTask, const int httpCode);
+        virtual void HandleResults(RequestTask& requestTask);
+
+        struct RequestImpl;
+        struct RequestTask
+        {
+            RequestTask();
+            virtual ~RequestTask();
+
+            bool Initialize(std::unique_ptr<CallRequestContainerBase>& requestContainer);
+            
+            enum State:int
+            {
+                None = 0,
+                Pending = RequestTask::None,
```

Comment:
>Pending = RequestTask::None [](start = 16, length = 27)

not sure why you need none, if you assign none here, both will have same value

Reply:
Not separated for the status of the value, but clearly for the programmer to recognize it when setting the value. This reduces mistakes in terms of maintenance.

Reply:
then why assign pending = none, instead intialize with none and then when you get the request mark it pending?

---
In reply to: [272404052](https://github.com/PlayFab/SDKGenerator/pull/464#discussion_r272404052) [](ancestors = 272404052)

Reply:
It is because the task created and added to pending queue immediatly. there is actually no state change on automata perspective.